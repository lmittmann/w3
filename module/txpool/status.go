package txpool

import (
	"encoding/json"

	"github.com/ethereum/go-ethereum/common/hexutil"
	"github.com/lmittmann/w3/internal/module"
	"github.com/lmittmann/w3/w3types"
)

// Status requests the txpool status.
func Status() w3types.CallerFactory[StatusResponse] {
	return module.NewFactory[StatusResponse](
		"txpool_status",
		nil,
	)
}

type StatusResponse struct {
	Pending uint
	Queued  uint
}

func (s *StatusResponse) UnmarshalJSON(input []byte) error {
	type statusResponse struct {
		Pending hexutil.Uint `json:"pending"`
		Queued  hexutil.Uint `json:"queued"`
	}

	var dec statusResponse
	if err := json.Unmarshal(input, &dec); err != nil {
		return err
	}
	s.Pending = uint(dec.Pending)
	s.Queued = uint(dec.Queued)
	return nil
}
