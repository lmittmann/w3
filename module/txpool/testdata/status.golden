> {"jsonrpc":"2.0","id":1,"method":"txpool_status"}
< {"jsonrpc":"2.0","id":1,"result":{"pending":"0xa","queued":"0x7"}}
